src/slashcommands/ISlashCommand.ts

```diff
@@ -0,0 +1,6 @@
+// just guessing
+export interface ISlashCommand {
+    command: string;
+    pattern: string;
+    function: () => boolean;
```

Comment:
Maybe this should be `executor(context: IExecutorContext): IExecutorResult;`?

Where the context contains the user and room along with the parameters passed to the command.

The result could contain a message to send, private notification to the user, etc.